include/openssl/ssl.h

```diff
@@ -2290,6 +2290,35 @@ __owur const struct openssl_ssl_test_functions *SSL_test_functions(void);
 __owur int SSL_free_buffers(SSL *ssl);
 __owur int SSL_alloc_buffers(SSL *ssl);
 
+typedef int TICKET_RETURN;
+
+/* Support for ticket appdata */
+/* fatal error, malloc failure */
+# define TICKET_FATAL_ERR_MALLOC 0
+/* fatal error, either from parsing or decrypting the ticket */
+# define TICKET_FATAL_ERR_OTHER  1
+/* No ticket present */
+# define TICKET_NONE             2
+/* Empty ticket present */
+# define TICKET_EMPTY            3
+/* the ticket couldn't be decrypted */
+# define TICKET_NO_DECRYPT       4
+/* a ticket was successfully decrypted */
+# define TICKET_SUCCESS          5
+/* same as above but the ticket needs to be renewed */
+# define TICKET_SUCCESS_RENEW    6
```

Comment:
I don't think we have any right to claim the `TICKET` namespace. This should probably be `SSL_TICKET_RETURN` or similar.